### ISSUE TITLE
davfs is not mounted via SpaceFM

### ISSUE THREAD

**Issue by Vladimir-csp** [IgnorantGuru/spacefm#648]
I've tried to mount davfs via SpaceFM, got this:

```
ROOT: /bin/mount -v -t davfs -o nosuid,noexec,nodev,uid=1000,gid=100 https://example.com/ /media/username/https-example.com
Please enter the username to authenticate with server
https://example.com/ or hit enter for none.
  Username: remoteusername
Please enter the password to authenticate user remoteusername with server
https://example.com/ or hit enter for none.
  Password:  
/sbin/mount.davfs: can't mount using fuse kernel file system
/sbin/mount.davfs: trying coda kernel file system
/sbin/mount.davfs: no free coda device to mount
      mount exit code = 1

Press Enter to close:
```

When mounting by udevil in terminal, everything works.
Both udevil from terminal and udevil from SpaceFM launch mount.davfs with identical arguments.

**Comment by Vladimir-csp** [comment]
I've refined davfs mount handler a bit to work with 'run in terminal' setting and whitelist:
`http https webdav davfs davs dav mtab_fs=davfs*`

```
#!/bin/bash

# Set your web browser in Help|Options|Browser
# set missing_davfs=1 if you always want to open http in web browser
# set missing_davfs=0 if you always want to mount http with davfs
missing_davfs=

if [ -z "$missing_davfs" ]; then
    grep -qs '^[[:space:]]*allowed_types[[:space:]]*=[^#]*davfs' /etc/udevil/udevil.conf 2>/dev/null
    missing_davfs=$?
fi

if [ $missing_davfs -eq 1 ] && [ "$fm_url_proto" -eq "http" -o "$fm_url_proto" -eq "https" ]; then
    # open in web browser
    spacefm -s run-task web "$fm_url"
    exit
fi

[ -n "$fm_url_user" ] && echo -e "\nPassing user in URL is not supported by davfs! Ignoring user.\n"
[ -n "$fm_url_password" ] && echo -e "\nPassing password in URL is not supported by davfs! Ignoring password.\n"

fm_url_proto="${fm_url_proto/webdav/http}"
fm_url_proto="${fm_url_proto/davfs/http}"
fm_url_proto="${fm_url_proto/davs/https}"
fm_url_proto="${fm_url_proto/dav/http}"

udevil mount "${fm_url_proto}://${fm_url_host}${fm_url_port:+:}${fm_url_port}${fm_url_path:-/}"
```

It mounts, informs about davfs poor URL handling in case of user or password are given in URL. But SpaceFM does not open mount point, despite using udevil.

**Comment by IgnorantGuru** [comment]
Thanks - the default davfs handler should now be fixed per your new bash code in next and alpha, due for 1.0.6 release.  If the user has changed the davfs handler, the new default will not be shown unless the Defaults button is clicked.

Testing is appreciated as I don't use davfs.  Also, the test for udevil.conf is imperfect, in that the user may be using a udevil-USER.conf file to enable davfs.  If so, davfs support may not be detected in udevil, in which case the handler will need to be edited.  Yet since that's an advanced use of udevil, I think it's good enough.

Also, regarding mtab_fs=fuse in the whitelist, it has been added to the default "fuse unmount" handler instead, which should take care of the unmount.  Also, the "fuse unmount" handler was not being added to the list of handlers correctly due to a false positive in the code which tests for its presence.  This has been corrected, so that handler should be added when upgrading, or one may click Options|Restore Default Handlers.

**Comment by IgnorantGuru** [comment]
Had some bad bash in there, should be fixed in f9df84f - wouldn't open a http link.

One thing I don't like about this solution is that it momentarily opens a terminal to open a regular http link in the user's web browser, which is why I used terminal conditionally in the original.  I think it would be better to only run the terminal for udevil if possible.
